spec/pipekit/request_spec.rb

```diff
@@ -62,28 +71,65 @@ module Pipekit
         fields = {"name" => "Bob", "middle_name" => "Milhous"}
         custom_field = Config.field("person", "middle_name")
 
-        stub = stub_post("persons?api_token=", "name=Bob&#{custom_field}=Milhous")
+        stub = stub_post("persons", "name=Bob&#{custom_field}=Milhous")
+
+        request.post(fields)
+
+        expect(stub).to have_been_requested
+      end
+    end
+
+    describe "#post" do
+      it "makes a post request to Pipedrive with the correct options" do
+        fields = {"name" => "Bob", "middle_name" => "Milhous"}
+        custom_field = Config.field("person", "middle_name")
+
+        stub = stub_post("persons", "name=Bob&#{custom_field}=Milhous")
 
         request.post(fields)
 
         expect(stub).to have_been_requested
       end
     end
 
-    def stub_get(uri, response)
-      stub_request(:get, "#{Pipekit::Request::PIPEDRIVE_URL}/#{uri}").to_return(status: 200, body: {success: true, data: response}.to_json)
+    describe "raising errors" do
+      it "raises an error when response is unsuccessful" do
+        stub_get("persons", "name=Bob", success: false)
+
+        expect{request.get}.to raise_error(UnsuccessfulRequestError) 
+      end
+
+      it "raises an error when resource is not found" do
+        stub_get("persons", "", success: true)
+
+        expect{request.get}.to raise_error(ResourceNotFoundError) 
+      end
+    end
+
+    def stub_get(uri, response, success: true, more_items: false, start: 0)
+      stub_request(:get, "#{Request::PIPEDRIVE_URL}/#{uri}?api_token=&limit=#{Request::DEFAULT_PAGINATION_LIMIT}&start=#{start}")
+        .to_return(status: 200, body: {
+          success: success,
+          data: response,
+          additional_data: {
+            pagination: {
+              more_items_in_collection: more_items,
+              next_start: Request::DEFAULT_PAGINATION_LIMIT
+            }
+          }
+        }.to_json)
     end
 
     def stub_put(uri, body)
-      stub_request(:put, "#{Pipekit::Request::PIPEDRIVE_URL}/#{uri}")
+      stub_request(:put, "#{Request::PIPEDRIVE_URL}/#{uri}?api_token=")
         .with(body: body)
-        .to_return(status: 200, body: {success: true}.to_json)
+        .to_return(status: 200, body: {success: true, data: "not empty"}.to_json)
     end
 
     def stub_post(uri, body)
-      stub_request(:post, "#{Pipekit::Request::PIPEDRIVE_URL}/#{uri}")
+      stub_request(:post, "#{Request::PIPEDRIVE_URL}/#{uri}?api_token=")
         .with(body: body)
-        .to_return(status: 200, body: {success: true}.to_json)
+        .to_return(status: 200, body: {success: true, data: "not empty"}.to_json)
```

Comment:
We could consider using VCR to use actual response data as a fixture. Could make it a bit easier to debug pipedrive weirdness.